src/ElmFormat/Filesystem.hs

```diff
@@ -2,10 +2,20 @@ module ElmFormat.Filesystem where
 
 import Control.Monad.Free
 import ElmFormat.FileStore
-import System.FilePath ((</>))
+import ElmVersion
+import System.FilePath ((</>), takeDirectory)
+
+import qualified Data.Maybe as Maybe
 import qualified System.FilePath as FilePath
 
 
+data ElmFile
+    = ElmFile
+        { version :: ElmVersion
+        , path :: FilePath
+        } deriving (Show)
```

Comment:
Just used `deriving Show` for my own private tests. It can be removed if you want.